### ISSUE TITLE
Bug: grady countercharged displays as checky

### ISSUE THREAD

**Issue by steve-the-player** [drawshield/Drawshield-Code#679]
When I ask it for a grady countercharged (using a grady of sable and white for an example), the countercharged ordinary behaves as if the original pattern was checky. I have tested this with multiple types of ordinaries and multiple directions of gradient, and it appears to be consistently bugged across all of them
![shield-grady_in_annulo-chevron_countercharged-BROKEN](https://github.com/drawshield/Drawshield-Code/assets/61434698/32adb42f-f7aa-41fa-a6d6-ffb5fadb3f6d)
_[Grady in annulo sable argent, a chevron countercharged](https://drawshield.net/create/index.html?blazon=grady%20in%20annulo%20sable%20argent%2C%20a%20chevron%20(grady%20in%20annulo%20argent%20sable)%20countercharged)_

Manually overwriting the blazon results in something like this 
![shield-grady_in_annulo-chevron_countercharged-manual_override-NOTQUITERIGHT](https://github.com/drawshield/Drawshield-Code/assets/61434698/6896e2f7-6506-413a-b867-7ac9f65e93ab)
_[Grady in annulo sable argent, a chevron grady in annulo argent sable](https://drawshield.net/create/index.html?blazon=grady%20in%20annulo%20sable%20argent%2C%20a%20chevron%20grady%20in%20annulo%20argent%20sable%20(countercharged))_
which is a lot closer, but still not quite right. Notably, when using a _grady in annulo_, or radial gradient, the center point of the chevron's gradient is lower than that of the field. When countercharging, it should merely appear to swap the two colors (or in this case reverse the direction of the gradient).

**Comment by karlwilcox** [comment]
Interesting. The grady thing was just a random idea had (not at all heraldic) so I didn't fully implement it. I think what is happening is that counterchanged looks for a two colour field, but grady isn't one (because you can have grady of a single colour), hence it falls back to checky ang gives an error message.

I could add some additional code to ensure the match, but then you run into the second problem that you note, which is that the centre of the circular fill is the center of the filled object's bounding box (which in most cases is what you actually want, e.g. on a roundel), but with counterchanged you want the centre to be in exactly the same place as the underlying field, and then use the object as a mask instead of filling it.

So, in short, yes, that is a bug, sorry. However, fixing it is probably quite fiddly (and still not heraldic) so to be honest, I'm probably not going to have time to look at this but I'm happy to leave it on the issues list for now.

Thanks for getting in touch!

**Comment by steve-the-player** [comment]
Yeah, I don’t really consider it urgent. I just thought I’d make sure it was documented as something to do later.
